internal/run/convert.go

```diff
@@ -88,7 +146,27 @@ func (c *Convert) Execute() error {
 		Context: c.kubeContext,
 	}
 
-	return convertcmd.Convert(c.convertOptions, kc)
+	clientset, err := clientsetFromFile(c.kubeConfig)
+	if err != nil {
+		return err
+	}
+
+	configmaps, err := c.getV2ReleaseConfigmaps(clientset)
+	if err != nil {
+		return err
+	}
+
+	if err := convertcmd.Convert(c.convertOptions, kc); err != nil {
+		return err
+	}
+
+	if !c.convertOptions.DeleteRelease {
+		if err := c.preserveV2ReleaseConfigmaps(clientset, configmaps, "converted-to-helm3"); err != nil {
+			return err
+		}
+	}
```

Comment:
```suggestion
	if !c.convertOptions.DeleteRelease {
		clientset, err := clientsetFromFile(c.kubeConfig)
		if err != nil {
			return err
		}

		configmaps, err := c.getV2ReleaseConfigmaps(clientset)
		if err != nil {
			return err
		}

		if err := convertcmd.Convert(c.convertOptions, kc); err != nil {
			return err
		}

		if err := c.preserveV2ReleaseConfigmaps(clientset, configmaps, "converted-to-helm3"); err != nil {
			return err
		}
	} else {
		if err := convertcmd.Convert(c.convertOptions, kc); err != nil {
			return err
		}
	}
```

Reply:
Is it better to move configmaps to the constructor if the releases are not being deleted? 

Then the convert branching is more clear with a single condition to preserve after convert.

Reply:
This avoids making unnecessary API calls. However that is achieved is fine by me.

Reply:
I want to have the option to delete releases after convert, and that's handled with the DeleteRelease flag. If Convert runs only when DeleteRelease is false then that option is not available

Reply:
> If Convert runs only when DeleteRelease is false

I don't think anyone is proposing this.

Reply:
sorry, missed the "else" clause at the end